torba/baseaccount.py

```diff
@@ -291,16 +297,19 @@ def get_details(self, show_seed=False, **kwargs):
 
     def decrypt(self, password):
         assert self.encrypted, "Key is not encrypted."
-        secret = double_sha256(password)
-        self.seed = aes_decrypt(secret, self.seed)
-        self.private_key = from_extended_key_string(self.ledger, aes_decrypt(secret, self.private_key))
+        self.seed = aes_decrypt(password, self.seed.encode()).decode()
+        self.private_key = from_extended_key_string(
+            self.ledger, aes_decrypt(password, self.private_key.encode()).decode()
+        )
+        self.password = password
         self.encrypted = False
 
     def encrypt(self, password):
         assert not self.encrypted, "Key is already encrypted."
-        secret = double_sha256(password)
-        self.seed = aes_encrypt(secret, self.seed)
-        self.private_key = aes_encrypt(secret, self.private_key.extended_key_string())
+        self.seed = aes_encrypt(password, self.seed.encode()).decode()
+        private_key: PrivateKey = self.private_key
```

Comment:
`self.private_key` should already have a type. this seems like the wrong place to declare the type. you can try forcing the type in the `__init__` if it's not already inferred.